### ISSUE TITLE
alpine image security updates/upgrades

### ISSUE THREAD

**Issue by petolexa** [Apicurio/apicurio-registry#1603]
Hi,

we started to store images in [Harbor](https://goharbor.io/) in our company internally (because k8s clusters do not have externall network access).
And one of Harbor functionalities is, that it scans vulnerabilities. In case of apicurio-registry-kafkasql 2.0.1.Final docker image it means this:
![obrazek](https://user-images.githubusercontent.com/11602126/123091832-191d5600-d42a-11eb-8122-a2c3a9debcb1.png)

All vulnerabilities are fixed in further updates of Alpine distro, but the image used for apicurio-registry images seems to be [discontinued for a few months](https://fabric8.io/).

So I wanted to ask few questions about it:
- Is there a high chance that we can break something running `apk upgrade` in the image? I mean, do you preserve this specific image for a specific reason?
- Is there a chance that you can add `apk upgrade` to the image packaging process?

Thank you,
Peter

**Comment by EricWittmann** [comment]
You should be fine upgrading.  I do not think we are pinned to a specific version for any reason - likely we just haven't refreshed it since we created the docker image originally.

@riprasad what do you think about upgrading our upstream docker images to some more recent to address CVEs?

**Comment by riprasad** [comment]
@petolexa You'd be glad to know that we have moved away from using the deprecated image for building our docker images. We now use the base image from the  [Red Hat Container Catalogue](https://catalog.redhat.com/software/containers/search), where images are continuously monitored for health and updates are pushed to address the known vulnerabilities and exposures.

![Screenshot from 2021-06-29 14-52-37](https://user-images.githubusercontent.com/61964599/123773242-25058e00-d8ea-11eb-9d24-15f7ac2ad4e2.png) 

You can have a look at the updated Dockerfile for kafkasql [here](https://github.com/Apicurio/apicurio-registry/blob/master/distro/docker/src/main/docker/Dockerfile.kafkasql.jvm#L1).

@EricWittmann I probably missed applying the changes to `2.0.x` branch. I'll raise a PR for the same and probably work on releasing upgraded images.

**Comment by petolexa** [comment]
I [should have linked the code](https://github.com/Apicurio/apicurio-registry/blob/14ddaebb074f2e26a35bdea23a0b96fdf69e122d/distro/docker/src/main/docker/Dockerfile.kafkasql.jvm#L1) in my original message, that shows the image used for this specific 2.0.1.Final version, my bad.

Thank you both for your detailed answers and for positive information.

**Comment by riprasad** [comment]
@petolexa We have upgraded the images for `2.0.1.Final`. Could you update the latest image in Harbour and check the vulnerability severity. I would be particularly interested in knowing what Harbour has to report for these new images.

**Comment by petolexa** [comment]
Hi @riprasad, it seems, that image for 2.0.1.Final in dockerhub is still on alpine - even if I see update from yesterday, if I run the image, I see:
```
/ # cat /etc/os-release
NAME="Alpine Linux"
ID=alpine
VERSION_ID=3.11.6
PRETTY_NAME="Alpine Linux v3.11"
BUG_REPORT_URL="https://bugs.alpinelinux.org/"
```
The size corresponds to alpine as well (cca 160MB comparing to UBI with cca 250MB).

So I tried latest-snapshot image and it shows RedHat:
```
[jboss@cf7d3b301646 ~]$ cat /etc/os-release
VERSION="8.4 (Ootpa)"
ID="rhel"
ID_LIKE="fedora"
VERSION_ID="8.4"
PLATFORM_ID="platform:el8"
PRETTY_NAME="Red Hat Enterprise Linux 8.4 (Ootpa)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:redhat:enterprise_linux:8.4:GA"
```

I pushed this latest snapshot to our Harbor and it has 0 critical and only 6 High severity issues ( comparing to 1+23 in that older Alpine):
![obrazek](https://user-images.githubusercontent.com/11602126/123940699-b7189f80-d999-11eb-85ac-2151596dd9d4.png)

From my point of view, it is good :) At least in the latest-snapshot. For 2.0.1.Final we are okay with the older Alpine, as we know, that the image for next versions will be more secure.

Thank you for your time,
Peter
